python/caffe/layers/detectnet/mean_ap.py

```diff
@@ -28,22 +26,31 @@ class 1 - true positive, 2 - false positive, 3 - true negative
         python_param {
             module: 'caffe.layers.detectnet.mean_ap'
             layer: 'ScoreDetections'
+            # parameter - IoU threshold (optional - default: 0.7)
+            param_str: '0.7'
         }
         include: { phase: TEST }
     }
     """
+    
 
     def setup(self, bottom, top):
-        pass
+        # Default IoU threshold
+        self.iou_threshold = 0.7 
+        try:
+            self.iou_threshold = float(self.param_str)
```

Comment:
I think it's a good idea. I would prefer `json` style instead of the string. We could even maintain backwards compatibility with something like:

```python
try:
    params = json.loads(self.param_str)
    self.iou_threshold = params.get('iou_threshold', self.iou_threshold)
except ValueError:
    try:
        self.iou_threshold = float(self.param_str)
    except:
        pass
```

The same for the `ClusterGroundtruth` and `ClusterDetections` layers. Do you think backwards compatibility is necessary? By the way, I understand the _Cons_, but I don't see any problem of not matching the format for other layers, given that we are adding relevant information.